src/java.base/share/classes/java/net/Socket.java

```diff
@@ -531,6 +531,8 @@ void createImpl(boolean stream) throws SocketException {
         try {
             impl.create(stream);
             created = true;
+        } catch (SocketException e) {
+            throw e;
         } catch (IOException e) {
             throw new SocketException(e.getMessage());
```

Comment:
I can't immediately think of real cases where a non-SocketException is thrown when creating a SocketImpl but for consistency with the equivalent in ServerSocket then you could add the cause here too.